File: prob1.py
# Time Complexity :  O(n)
# Space Complexity :O(1)
# Passed on Leetcode: yes

class Solution:
    def canJump(self, nums: List[int]) -> bool:
        n = len(nums)
        if n == 1:
            return True
        target = n-1
        for i in range(n-2, -1, -1):
            if i + nums[i] >= target:
                target = i
        
        return target == 0

File: prob1.java
// Time Complexity : O(k * n) 
// Space Complexity : O(k)
// Did this code successfully run on Leetcode : Yes

class Solution {
    public int superEggDrop(int k, int n) {
       
       int dp[]=new int[k+1], m;
       for(m=0;dp[k] < n; m++){
            
            for(int x=k;x>0;x--){
                dp[x]+=1+dp[x-1];
            }
       }
       return m;
    }
}